src/main/java/randoop/resource/ClassAnnotationScanner.java

```diff
@@ -0,0 +1,149 @@
+import org.objectweb.asm.ClassVisitor;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.TypePath;
+import org.plumelib.reflection.Signatures;
+
+/**
+ * ClassAnnotationScanner is a ClassVisitor that collects methods that are annotated with specific
+ * annotations.
+ */
+public class ClassAnnotationScanner extends ClassVisitor {
+  /** OpCode for ASM used throughout this class. */
+  private static final int ASM_OPCODE = Opcodes.ASM7;
+
+  /** The desired annotations that serve as the criteria for methods we want to capture. */
+  private final Collection<String> desiredAnnotations;
+
+  /** The binary name of the class being visited. */
+  private @BinaryName String className = null;
+
+  /** The accumulated set of captured fully qualified method signatures. */
+  private final Set<String> matchingFullyQualifiedMethodSignatures = new HashSet<>();
+
+  /**
+   * Initializes a new AnnotationScanner to look for methods with the specified annotations.
+   *
+   * @param api API version for ASM
+   * @param desiredAnnotations annotations to look for
+   */
+  public ClassAnnotationScanner(int api, Collection<String> desiredAnnotations) {
+    super(api);
+    this.desiredAnnotations = desiredAnnotations;
+  }
+
+  /**
+   * Returns the Randoop fully qualified signature of a method. e.g. "java.util.String.hashCode()"
+   *
+   * @param method method name
+   * @param argumentSignature signature of the method's arguments, e.g.
+   *     "(char[],int,int,java.lang.String,int)"
+   * @return Randoop fully qualified signature
+   */
+  private String getFullyQualifiedMethodSignature(String method, String argumentSignature) {
+    return className + "." + method + argumentSignature;
+  }
+
+  /**
+   * Adds a method to the set of captured methods if any annotation on or within the method
+   * signature is in {@code desiredAnnotations}. Ignores constructors and static constructors.
+   *
+   * @param annotationName annotation name, e.g.
+   *     "org.checkerframework.checker.determinism.qual.NonDet"
+   * @param method method name
+   * @param argumentSignature signature of the method's arguments, e.g.
+   *     "(char[],int,int,java.lang.String,int)"
+   */
+  private void addMethodIfMatching(String annotationName, String method, String argumentSignature) {
+    if (method.contains("<init>") || method.contains("<clinit>")) {
+      return; // Ignore constructors and static constructors
+    }
+
+    if (desiredAnnotations.contains(
+        Signatures.fieldDescriptorToBinaryName(annotationName.trim()))) {
+      matchingFullyQualifiedMethodSignatures.add(
+          getFullyQualifiedMethodSignature(method, argumentSignature));
+    }
+  }
+
+  @Override
+  public void visit(
+      int version,
+      int access,
+      java.lang.String name,
+      java.lang.String signature,
+      java.lang.String superName,
+      java.lang.String[] interfaces) {
+    className = name.replace('/', '.');
```

Comment:
What is the format of `name`?  It's better to use a method from the [`Signatures`](http://plumelib.org/reflection-util/api/org/plumelib/reflection/Signatures.html) class rather than doing your own string replacement.